src/ninja.cc

```diff
@@ -845,6 +846,76 @@ int NinjaMain::ToolInputs(const Options* options, int argc, char* argv[]) {
   return 0;
 }
 
+int NinjaMain::ToolMultiInputs(const Options* options, int argc, char* argv[]) {
+  // The inputs tool uses getopt, and expects argv[0] to contain the name of
+  // the tool, i.e. "inputs".
+  argc++;
+  argv--;
+
+  bool print0 = false;
+
+  optind = 1;
+  int opt;
+  const char* defaultfieldseparator = "\t";
+  const char* fieldseparator = defaultfieldseparator;
+  const option kLongOptions[] = { { "help", no_argument, NULL, 'h' },
+                                  { "fieldseparator", required_argument, NULL,
+                                    'F' },
+                                  { "print0", no_argument, NULL, '0' },
+                                  { NULL, 0, NULL, 0 } };
+  while ((opt = getopt_long(argc, argv, "F:h0", kLongOptions, NULL)) != -1) {
+    switch (opt) {
+    case 'F':
+      fieldseparator = optarg;
+      break;
+    case '0':
+      print0 = true;
+      break;
+    case 'h':
+    default:
+      // clang-format off
+      printf(
+"Usage '-t multiinputs [options] [targets]\n"
+"\n"
+"List all target+inputs used for a set of targets, sorted in dependency order.\n"
```

Comment:
nit: clarify a bit what the expected output is here too. In particular desscribe the output format as a series of lines, each one like `<target> <delimiter> <input> <terminator>`

Reply:
Updated... hopefully for the better.